src/string.cr

```diff
@@ -3314,7 +3314,7 @@ class String
           if mem
             if char == '_'
               # case 3
-            else
+            elsif upcase || downcase
```

Comment:
Yup, I saw that Rails behaves this way too ...

Anyway, I'm not sure it's consistent in Rails either, [the doc](http://api.rubyonrails.org/classes/ActiveSupport/Inflector.html#method-i-underscore) says `As a rule of thumb you can think of underscore as the inverse of camelize` so in that case, the camelized version of "123abc" should be "123Abc" but it is "123abc".

That's a lot of discussions for such a small problem :)
I prefer the way it's done in Rails but it feels strange in some ways too ...

(Sorry again for the double PR, I'll close this one as soon as we have an answer from the devs)